model_api/python/model_api/models/instance_segmentation.py

```diff
@@ -199,14 +195,17 @@ def postprocess(self, outputs, meta):
             saliency_maps = [[] for _ in range(len(self.labels))]
         else:
             saliency_maps = []
-        for box, confidence, cls, str_label, raw_mask in zip(
-            boxes, scores, labels, str_labels, masks
-        ):
+        for box, confidence, cls, raw_mask in zip(boxes, scores, labels, masks):
             x1, y1, x2, y2 = box
             if (x2 - x1) * (y2 - y1) < 1 or (
                 confidence <= self.confidence_threshold and not has_feature_vector_name
             ):
                 continue
+            str_label = (
+                self.labels[cls]
+                if self.labels and cls < len(self.labels)
+                else f"#{cls}"
```

Comment:
Why do we need to generate predictions which are not aligned with the labels we have in rt_info?

Reply:
@sovrasov We could skip it by

```py
for box, confidence, cls, raw_mask in zip(boxes, scores, labels, masks):
    x1, y1, x2, y2 = box
    area = (x2 - x1) * (y2 - y1)

    if area < 1 or (confidence <= self.confidence_threshold and not has_feature_vector_name):
        continue

    # Skip if label index is out of bounds
    if self.labels and cls >= len(self.labels):
        continue

    # Get label string
    str_label = self.labels[cls] if self.labels else f"#{cls}"
```

Reply:
That looks better, and would allow us to prevent assigning undefined labels